client/components/domains/contact-details-form-fields/index.jsx

```diff
@@ -144,8 +146,6 @@ export class ContactDetailsFormFields extends Component {
 		const mainFieldValues = formState.getAllFieldValues( this.state.form );
 		return {
 			...mainFieldValues,
-			// domains registered according to ancient validation rules may have state set even though not required
-			state: this.props.hasCountryStates ? mainFieldValues.state : '',
```

Comment:
Totally makes no sense to blank the value if there are no country states. We still allow a bunch of text input states, and I'm pretty sure these would fail if we blanked them. (or sent blanks initially)